companion/src/mainwindow.cpp

```diff
@@ -1398,21 +1400,35 @@ void MainWindow::updateProfilesActions()
   }
 }
 
-void MainWindow::createProfile()
+
+void MainWindow::copyProfile()
 {
   int i;
   for (i=0; i<MAX_PROFILES && g.profile[i].existsOnDisk(); i++);
   if (i==MAX_PROFILES)  //Failed to find free slot
     return;
 
-  // Copy current profile to new and give it a new name
   g.profile[i] = g.profile[g.id()];
   g.profile[i].name(tr("New Radio"));
 
   g.id(i);
   updateMenus();
 }
 
+void MainWindow::createProfile()
+{
+  int i;
+  for (i=0; i<MAX_PROFILES && g.profile[i].existsOnDisk(); i++);
+  if (i==MAX_PROFILES)  //Failed to find free slot
+    return;
+
+  g.profile[i].init(i);
+  g.profile[i].name(tr("New Radio"));
+
+  g.id(i);
+  updateMenus();
+}
+
```

Comment:
This is cool, I like the new option for adding vs. copying.  👍 

But shouldn't `copyProfile()` first call `createProfile()` and then just copy the settings from the old one to new?  To avoid duplicating all that code.